### PR TITLE
OWNERS: update TechDocs + add Discoverability

### PR DESCRIPTION
**Pull request by Rugvip**
As the original TechDocs team has taken on broader responsibility of Discovery within Backstage, focused has been shifting away from TechDocs, which is in maintenance mode for the time being.

This forms the "Discoverability" project area, where the goal of helping end users find what they are looking for within Backstage. This includes making sure we have a delightful landing page experience, solid search in Backstage itself, and an information architecture that makes it easy to browse Backstage.

In addition to this change, the `@backstage/techdocs-core` team will be renamed to `@backstage/techdocs-maintainers`, and the new `@backstage/discoverability-maintainers` will be created.

^ describes the proposed change, please chime in with any feedback 😁

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -71,13 +71,13 @@ yarn.lock                                           @backstage/maintainers @back
...
-/plugins/search-*                                   @backstage/techdocs-core
+/plugins/search                                     @backstage/discoverability-maintainers
+/plugins/search-*                                   @backstage/discoverability-maintainers
 /plugins/sonarqube                                  @backstage/maintainers @backstage/sda-se-reviewers
-/plugins/stack-overflow                             @backstage/techdocs-core
-/plugins/stack-overflow-backend                     @backstage/techdocs-core
-/plugins/techdocs                                   @backstage/techdocs-core
-/plugins/techdocs-*                                 @backstage/techdocs-core
+/plugins/stack-overflow                             @backstage/discoverability-maintainers
+/plugins/stack-overflow-backend                     @backstage/discoverability-maintainers
```
Comment (Rugvip) [inline .github/CODEOWNERS:78]: Should this be here?

**Review by emmaindal — APPROVED** [review]
This looks good from my perspective, will let @soapraj have a final look as well!

File: .github/CODEOWNERS (hunk 1 above)
Comment (emmaindal) [reply .github/CODEOWNERS:78]: For now I would say yes!

File: OWNERS.md [hunk 2]
```diff
@@ -38,17 +52,11 @@ Scope: The Kubernetes plugin and the base it provides for other plugins to build
...
-| TBD  | Spotify      | Pulp Fiction | [TBD](http://github.com/TBD) | -       |
-
-### Search
-
-Team: @backstage/search-maintainers
+Scope: The TechDocs plugin and related tooling
 
-| Name | Organization | Team | GitHub                       | Discord |
-| ---- | ------------ | ---- | ---------------------------- | ------- |
-| TBD  | Spotify      | BUX  | [TBD](http://github.com/TBD) | -       |
```
Comment (soapraj) [inline OWNERS.md:51]: Can you add me here as well for the time being?

**Review by benjdlambert — APPROVED** [review]

**Review by jhaals — APPROVED** [review]

**Pull request merged**